connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java

```diff
@@ -20,13 +20,45 @@
 import org.apache.kafka.clients.producer.RecordMetadata;
 
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * SourceTask is a Task that pulls records from another system for storage in Kafka.
  */
 public abstract class SourceTask implements Task {
 
+    /**
+     * <p>
+     * The configuration key that determines how source tasks will define transaction boundaries
+     * when exactly-once support is enabled.
+     * </p>
+     */
+    public static final String TRANSACTION_BOUNDARY_CONFIG = "transaction.boundary";
+
+    public enum TransactionBoundary {
+        POLL,
+        INTERVAL,
+        CONNECTOR;
+
+        public static final TransactionBoundary DEFAULT = POLL;
+
+        public static List<String> options() {
```

Comment:
Neither of these patterns play nicely with the validators offered by `ConfigDef`, though. Both `ValidString` and `CaseInsensitiveValidString` require an array (or really, a varargs list) of strings in their constructors, which prevents us from using `Enum::values` directly to create one of these validators.

I've removed `options()` (from both `TransactionBoundary` and `ExactlyOnceSupportLevel`) in favor of a reusable `Utils::enumOptions` method, which returns a `String[]` containing the names (retrieved via `toString`) of each value for an enum class. This prevents us from having to duplicate this logic and allows us to use it without adding new methods to public API.

The actual implementations for both enums now follow the `ConnectorType` pattern, with an overridden `toString` method that returns the lowercase name of the enum and a static `fromProperty` method (which mirrors the `ConnectorType::fromValue` method) that parses a string case-insensitively into a value for the enum. I opted for a different method name for the latter since the values given to `fromProperty` should always be user-supplied values and the motivation for this separate method (instead of just invoking `Enum::valueOf` directly) is directly related to parsing user-supplied values.

We could also add a general-purpose `EnumValidator` class to `ConfigDef` (for public use) or in a section of the Connect code base reserved for private API, but that would still leave us with the trouble of having to re-implement logic to follow the phrase "Permitted values are" in property docstrings, and it's probably best to hold off on implementing a public-facing validator like that until its exact behavior can be agreed on with a KIP.